src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/inbound/InboundUserAcceptanceTest.java

```diff
@@ -82,52 +103,79 @@ void testSendEdifactIsProcessedAndPushedToGpOutboundQueue()
-        final List<String> messageIds = waitFor(() -> {
-            final List<String> inboxMessageIds = labResultMeshClient.getInboxMessageIds();
-            return inboxMessageIds.isEmpty() ? null : inboxMessageIds;
-        });
-        var meshMessage = labResultMeshClient.getEdifactMessage(messageIds.get(0));
-
-        //TODO NIAD-851 UAT framework
-//        Interchange expectedNhsAck = edifactParser.parse(nhsAck);
-//        Interchange actualNhsAck = edifactParser.parse(meshMessage.getContent());
-//
-//        assertThat(meshMessage.getWorkflowId())
-//            .isEqualTo(WorkflowId.PATHOLOGY_ACK);
-//        assertThat(actualNhsAck.getInterchangeHeader().getRecipient())
-//            .isEqualTo(expectedNhsAck.getInterchangeHeader().getRecipient());
-//        assertThat(actualNhsAck.getInterchangeHeader().getSender())
-//            .isEqualTo(expectedNhsAck.getInterchangeHeader().getSender());
-//        assertThat(actualNhsAck.getInterchangeHeader().getSequenceNumber())
-//            .isEqualTo(expectedNhsAck.getInterchangeHeader().getSequenceNumber());
-//        assertThat(filterTimestampedSegments(actualNhsAck))
-//            .containsExactlyElementsOf(filterTimestampedSegments(expectedNhsAck));
-//        assertThat(actualNhsAck.getInterchangeTrailer().getNumberOfMessages())
-//            .isEqualTo(expectedNhsAck.getInterchangeTrailer().getNumberOfMessages());
-//        assertThat(actualNhsAck.getInterchangeTrailer().getSequenceNumber())
-//            .isEqualTo(expectedNhsAck.getInterchangeTrailer().getSequenceNumber());
+    /**
+     * The test EDIFACT message (.dat file) is sent to the MESH mailbox where the adaptor receives inbound messages.
+     * The test waits for the messages to be processed confirms that the outbound queue is empty.
+     */
+    @ParameterizedTest(name = "[{index}] - {0}")
+    @ArgumentsSource(FailureArgumentsProvider.class)
+    void testEdifactFailsToBeProcessedAndNothingPushedToGpOutboundQueue(String category, TestData testData) {
+
+        String recipient;
+        try {
+            recipient = getEdifactParser().parse(testData.getEdifact())
+                .getInterchangeHeader().getRecipient();
+        } catch (InterchangeParsingException e) {
+            recipient = e.getRecipient();
+        } catch (MessageParsingException e) {
+            recipient = e.getRecipient();
+        }
+
+        WorkflowId workflowId;
+        if (testData.getEdifact().contains(WorkflowId.PATHOLOGY.getWorkflowId())) {
+            workflowId = WorkflowId.PATHOLOGY;
+        } else if (testData.getEdifact().contains(WorkflowId.SCREENING.getWorkflowId())) {
+            workflowId = WorkflowId.SCREENING;
+        } else {
+            throw new RuntimeException("Unsupported Workflow ID");
+        }
+
+        final boolean ackRequested = testData.getEdifact().replace("\n", "")
+            .matches(ACK_REQUESTED_REGEX);
```

Comment:
As before if applicable:
```suggestion
        final boolean ackRequested = testData.getEdifact()
            .matches(ACK_REQUESTED_REGEX);
```

Reply:
Adjusted this as before.